### DISCUSSION TITLE
How does the input system work?

### DISCUSSION THREAD

**Discussion by test3211234** [sxyazi/yazi#1278] [1. Q&A]
I was looking into the code to take inspiration from but it's so confusing. Could the creator explain how the entire keybinding and input and calling the right functions based off that, etc etc etc system works?

@sxyazi , I'm also wondering if you think a general crate for this kind of thing would be useful.

**Comment by AnirudhG07** [comment]
> I was looking into the code to take inspiration from but it's so confusing. Could the creator explain how the entire keybinding and input and calling the right functions based off that, etc etc etc system works?

https://yazi-rs.github.io/docs/configuration/keymap
The docs has various configs of input/output, keybindings. Also Check out various plugins, their codes are short and easy to understand. You can check their keybindings to get an idea of whats going on. 
I hope this helps a bit.
